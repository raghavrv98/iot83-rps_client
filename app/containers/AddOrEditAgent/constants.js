/*
 *
 * AddOrEditAgent constants
 *
 */

export const CREATE_AGENT_REQUEST = "app/AddOrEditAgent/CREATE_AGENT_REQUEST";
export const CREATE_AGENT_SUCCESS = "app/AddOrEditAgent/CREATE_AGENT_SUCCESS";
export const CREATE_AGENT_FAILURE = "app/AddOrEditAgent/CREATE_AGENT_FAILURE";

export const GET_AGENT_DETAILS = "app/AddOrEditAgent/GET_AGENT_DETAILS";
export const GET_AGENT_DETAILS_SUCCESS = "app/AddOrEditAgent/GET_AGENT_DETAILS_SUCCESS";
export const GET_AGENT_DETAILS_FAILURE = "app/AddOrEditAgent/GET_AGENT_DETAILS_FAILURE";
