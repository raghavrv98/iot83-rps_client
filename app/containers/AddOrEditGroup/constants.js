/*
 *
 * AddOrEditGroup constants
 *
 */

export const ON_SUBMIT_REQUEST = "app/AddOrEditGroup/ON_SUBMIT_REQUEST";
export const ON_SUBMIT_REQUEST_SUCCESS = "app/AddOrEditGroup/ON_SUBMIT_REQUEST_SUCCESS";
export const ON_SUBMIT_REQUEST_FAILURE = "app/AddOrEditGroup/ON_SUBMIT_REQUEST_FAILURE";

export const GET_GROUP_DETAILS = "app/AddOrEditGroup/GET_GROUP_DETAILS";
export const GET_GROUP_DETAILS_SUCCESS = "app/AddOrEditGroup/GET_GROUP_DETAILS_SUCCESS";
export const GET_GROUP_DETAILS_FAILURE = "app/AddOrEditGroup/GET_GROUP_DETAILS_FAILURE";
