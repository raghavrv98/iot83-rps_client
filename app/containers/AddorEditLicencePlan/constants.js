/*
 *
 * AddorEditLicencePlan constants
 *
 */

export const DEFAULT_ACTION = "app/AddorEditLicencePlan/DEFAULT_ACTION";

export const ON_SUBMIT_REQUEST = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST";
export const ON_SUBMIT_REQUEST_SUCCESS = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST_SUCCESS";
export const ON_SUBMIT_REQUEST_FAILURE = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST_FAILURE";

export const GET_PLANS_LIST = "app/AddorEditLicencePlan/GET_PLANS_LIST";
export const GET_PLANS_LIST_SUCCESS = "app/AddorEditLicencePlan/GET_PLANS_LIST_SUCCESS";
export const GET_PLANS_LIST_FAILURE = "app/AddorEditLicencePlan/GET_PLANS_LIST_FAILURE";