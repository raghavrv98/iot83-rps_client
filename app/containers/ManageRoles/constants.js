/*
 *
 * ManageRoles constants
 *
 */

export const DEFAULT_ACTION = "app/ManageRoles/DEFAULT_ACTION";

export const GET_LIST = "app/ManageRoles/GET_LIST";
export const GOT_LIST = "app/ManageRoles/GOT_LIST";
export const NOT_GOT_LIST = "app/ManageRoles/NOT_GOT_LIST";

export const DELETE_REQUEST = "app/ManageRoles/DELETE_REQUEST";
export const DELETE_REQUEST_SUCCESS = "app/ManageRoles/DELETE_REQUEST_SUCCESS";
export const DELETE_REQUEST_FAILURE = "app/ManageRoles/DELETE_REQUEST_FAILURE";