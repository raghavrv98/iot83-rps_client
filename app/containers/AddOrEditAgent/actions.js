/*
 *
 * AddOrEditAgent actions
 *
 */

import { CREATE_AGENT_REQUEST,GET_AGENT_DETAILS } from "./constants";

export function createAgent(data) {
  return {
    type: CREATE_AGENT_REQUEST,
    payload: data,
  };
}
export function getAgentDetails(id) {
  return {
    type: GET_AGENT_DETAILS,
    id
  };
}
