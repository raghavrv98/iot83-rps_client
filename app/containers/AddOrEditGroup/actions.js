/*
 *
 * AddOrEditGroup actions
 *
 */

import { ON_SUBMIT_REQUEST,GET_GROUP_DETAILS } from "./constants";

export function onSubmitHandler(payload,id) {
  return {
    type: ON_SUBMIT_REQUEST,
    payload,
    id
  };
}

export function getGroupDetails(id) {
  return {
    type: GET_GROUP_DETAILS,
    id
  };
}
