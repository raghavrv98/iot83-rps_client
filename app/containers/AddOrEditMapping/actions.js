/*
 *
 * AddOrEditMapping actions
 *
 */

import { DEFAULT_ACTION, GET_PLANTS,GET_PIPELINES,GET_EDIT_SAVE_MAP,GET_MAPPING_DATA,GET_AGENT_DETAILS } from "./constants";

export function defaultAction() {
  return {
    type: DEFAULT_ACTION
  };
}

export function getPlantList() {
  return {
    type: GET_PLANTS
  };
}

export function handelMapping(payload,expId,mapId) {
  return {
    type: GET_EDIT_SAVE_MAP,
    payload,
    expId,
    mapId
  };
}

export function getmappingDetails(agentId,mapId) {
  return {
    type: GET_MAPPING_DATA,
    agentId,mapId
  };
}

export function getAgentDetails(id) {
  return {
    type: GET_AGENT_DETAILS,
    id
  };
}