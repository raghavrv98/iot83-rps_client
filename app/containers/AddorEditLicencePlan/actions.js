/*
 *
 * AddorEditLicencePlan actions
 *
 */

import { DEFAULT_ACTION, ON_SUBMIT_REQUEST, GET_PLANS_LIST } from "./constants";

export function defaultAction() {
  return {
    type: DEFAULT_ACTION
  };
}

export function onSubmitHandler(payload) {
  return {
    type: ON_SUBMIT_REQUEST,
    payload
  };
}

export function getPlansList(){
  return{
    type: GET_PLANS_LIST
  }
}