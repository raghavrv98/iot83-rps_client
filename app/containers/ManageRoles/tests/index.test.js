import React from 'react';
import { shallow } from 'enzyme';
// import { enzymeFind } from 'styled-components/test-utils';

import { mapDispatchToProps, ManageRoles } from '../index';
import { getRolesList } from '../actions'
import { GET_LIST ,DELETE_REQUEST} from "../constants";
import history from 'utils/history';
const mockGroupList = [
  {
    "id": "9f1c2f19a97d4a01bebec08ddd206fe2",
    "name": "DEFAULT",
    "description": "Default container",
    "mqttConnectionId": null,
    "devices": 0
  }
]

describe('<ManageRoles />', () => {
  it('should test component when recieve props', () => {
    spyOn(ManageRoles.prototype, 'componentWillReceiveProps').and.callThrough();
    const wrapper = shallow(<ManageRoles getRolesList={getRolesList} list={undefined} listError={undefined} />);
    expect(wrapper).toBeDefined();
    wrapper.setProps({ list: mockGroupList, listError: "No Error Found." });
    expect(ManageRoles.prototype.componentWillReceiveProps).toHaveBeenCalledTimes(1);
  })

  // it("clicking add button trigger onClick event", async () => {
  //   let expectedCount = history.length + 1
  //   const wrapper = shallow(<ManageRoles getRolesList={getRolesList} history={history} />);
  //   let addButton = wrapper.find("button.btn-danger");
  //   expect(addButton.length).toBe(1);
  //   addButton.simulate('click');
  //   expect(history.length).toEqual(expectedCount);
  //   expect(history.location.pathname).toEqual("/addOrEditRole");
  // });

  it('Function testing on mapDispatchToProps', () => {
    const dispatch = jest.fn();
    const payload = "demo"
    const mockParams = {
      params: {
        id: "demoId"
      }
    }
    mapDispatchToProps(dispatch).getRolesList();
    expect(dispatch.mock.calls[0][0]).
    toEqual({ 
      type: GET_LIST,
    })
    mapDispatchToProps(dispatch).deleteHandler(mockParams.params.id);
    expect(dispatch.mock.calls[1][0]).
    toEqual({
      type: DELETE_REQUEST,
      id: "demoId"
    })
  });

  it('match snapshots', () => {
    const wrap = shallow(<ManageRoles getRolesList={getRolesList} />)
    expect(wrap).toMatchSnapshot()
  })
});
