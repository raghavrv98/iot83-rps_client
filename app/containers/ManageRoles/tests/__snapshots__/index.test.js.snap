// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ManageRoles /> match snapshots 1`] = `
<div
  className="appContent"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      ManageRoles
    </title>
    <meta
      content="Description of ManageRoles"
      name="description"
    />
  </HelmetWrapper>
  <div
    className="pageBreadcrumb"
  >
    <div
      className="flex-item fx-b70"
    >
      <p>
        <FormattedMessage
          defaultMessage="Identity Access Management"
          id="app.common.messages.iam"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
      </p>
      <h5>
        <FormattedMessage
          defaultMessage="All Available Roles"
          id="app.containers.ManageRoles.title"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
      </h5>
    </div>
    <div
      className="flex-item fx-b30 text-right align-items-center"
    />
  </div>
  <SkeletonLoader
    mode="fullView"
    skeleton="skeletonReactTable"
  />
</div>
`;
